src/types.rs

```diff
@@ -240,6 +240,65 @@ impl BlockTime {
     }
 }
 
+/// Balance differentiated in various categories
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Default)]
+pub struct Balance {
+    /// All coinbase outputs not yet matured
+    pub immature: u64,
+    /// Unconfirmed UTXOs generated by a wallet tx
+    pub trusted_pending: u64,
+    /// Unconfirmed UTXOs received from an external wallet
+    pub untrusted_pending: u64,
+    /// Confirmed and immediately spendable balance
+    pub available: u64,
```

Comment:
Would it make more sense to rename this as `confirmed`? Because, shouldn't `trusted_pending` be "available" in most cases?

Reply:
I like `confirmed`, but I'm wondering if then `untrusted_pending` is clear or not...

Reply:
Untrusted unconfirmed? Maybe a little too much 😅

I personally think that untrusted_pending is clear enough (given our options):

- Untrusted: we don't know if someone will replace it.

- Pending: Not confirmed.